crates/api/src/chronicle_graphql/mod.rs

```diff
@@ -560,23 +559,13 @@ impl async_graphql::extensions::Extension for OpaCheck {
                 .into_iter()
                 .collect(),
             );
-            let verdict = opa_executor
-                .lock()
-                .await
-                .evaluate(identity, &opa_context)
-                .await;
+            let verdict = opa_executor.evaluate(identity, &opa_context).await;
             match verdict {
-                Ok(true) => next.run(ctx, info).await,
-                Ok(false) => {
-                    tracing::warn!(
-                        "attempt to violate policy rules by {:?} in context {:#?}",
-                        identity,
-                        opa_context
-                    );
-                    Err(ServerError::new("violation of policy rules", None))
-                }
+                Ok(()) => next.run(ctx, info).await,
                 Err(error) => Err(ServerError {
-                    message: "cannot check policy rules".to_string(),
+                    message: format!(
+                        "attempt to violate policy rules by {identity} in context {opa_context:#?}"
```

Comment:
It seemed safer to me to keep the internal state out of the `ServerError::message` that gets sent to the remote user, instead just keeping it to the logs so only the admins see the server's details of attempted violations.

Reply:
Thanks for catching this!